packages/components/src/internal/renderers.tsx

```diff
@@ -98,169 +115,303 @@ export const EditableColumnTitle: FC<EditableColumnTitleProps> = memo(props => {
 
-    const _handleHideColumn = useCallback(() => {
-        setOpen(false);
-        handleHideColumn(col);
-    }, [col, handleHideColumn]);
+        setMenuStyle({
+            left,
+            // use visibility so we can know the rendered size of the menu before making it visible
+            visibility: open ? 'visible' : 'hidden',
+            top,
+        });
+    }, [open]);
 
-    const _handleAddColumn = useCallback(() => {
-        setOpen(false);
-        handleAddColumn(col);
-    }, [col, handleAddColumn]);
+    // In order to close the menu when the user clicks outside of it we have to add a click handler to the document and
+    // close the menu when the user clicks on anything outside of the menu.
+    const documentClickHandler = useCallback(
+        event => {
+            if (open && !menuEl.current.contains(event.target)) {
+                setOpen(false);
+            }
+        },
+        [setOpen, open]
+    );
 
-    const editColumnTitle = useCallback(() => {
-        setOpen(false);
-        setEditingTitle(true);
-        onColumnTitleEdit?.(col);
-    }, [col, onColumnTitleEdit]);
+    useEffect(() => {
+        document.addEventListener('click', documentClickHandler);
+        return () => {
+            document.removeEventListener('click', documentClickHandler);
+        };
+    }, [documentClickHandler]);
 
-    const onColumnTitleUpdate = useCallback(
-        (newTitle: string) => {
-            onColumnTitleChange(col.set('caption', newTitle) as QueryColumn);
-        },
-        [col, onColumnTitleChange]
+    // TODO: investigate passing down a ref of the .table-responsive div so we can add a scroll handler to it here the
+    //  same way we add one to the document, then we can update the menu positions when the table is also scrolled.
+    useEffect(() => {
+        updateMenuStyle();
+        window.addEventListener('scroll', updateMenuStyle);
+        return () => {
+            window.removeEventListener('scroll', updateMenuStyle);
+        };
+    }, [updateMenuStyle, open]);
+
+    // Technically we don't need to add and remove this open class because it doesn't affect visibility, we do that
+    // above via the visibility css property. We need this class so tests can look for the currently open menu.
+    const className = classNames('grid-header-cell__dropdown-menu dropdown-menu', { open });
+
+    const body = (
+        <ul className={className} ref={menuEl} style={menuStyle}>
+            {allowColFilter && (
+                <>
+                    <MenuItem onSelect={openFilterPanel}>
+                        <span className="fa fa-filter grid-panel__menu-icon" />
+                        Filter...
+                    </MenuItem>
+                    <MenuItem disabled={!colFilters || colFilters?.length === 0} onSelect={removeFilter}>
+                        <span className="grid-panel__menu-icon-spacer" />
+                        Remove filter{colFilters?.length > 1 ? 's' : ''}
```

Comment:
I believe the `&nbsp;` was here to fix the alignment for those menu items that don't have icons. Maybe the menu-icon-spacer padding just needs to be updated instead.
<img width="396" alt="Screen Shot 2022-10-27 at 9 12 21 AM" src="https://user-images.githubusercontent.com/6411206/198309509-af05369f-e3dd-48c2-b201-ba36d1c20cbc.png">

Reply:
Yeah, I'll swap it to add margin to the text instead of the icons.